NS-Proto-Appendix.md

```diff
@@ -258,3 +226,28 @@ Returns a `Tuple Iterator` object that contains the values for each index in the
 ## `Tuple.prototype.with(index, value)`
 
 Returns a `Tuple` with the same elements as the original `Tuple` except the element at `index` is replaced with `value`.
+
+# Compatibility with the existing `Object` methods
+
+> NOTE: Since `Record` doesn't inherit from `Object`, you can't directly call the `Object.prototype.*` methods but you have to use `.call()`.
+
+## `Object` methods which work with `Record` and `Tuple`
+
+- `Object.is()` - When comparing records and tuples it behaves exactly like `===`
+- `Object.entries()`, `Object.keys()`, `Object.values()` - The resulting arrays are sorted alphabetically by key
+- `Object.assign()` - Copy properties from a record or tuple to another object. **Caveat**: The first argument cannot be a record or a tuple, because they are immutable
+- `Object.isExtensible()`, `Object.isFrozen()`, `Object.isSealed()` - They always return `false` for records and tuples
+- `Object.preventExtension()`, `Object.freeze()`, `Object.seal()` - Records and tuples are already immutable, so these methods will not throw but won't have any effect
+- `Object.getOwnPropertyDescriptor()`, `Object.getOwnPropertyDescriptors()` - The resulting descriptors all have `writable: false, enumerable: true, configurable: false`
+- `Object.getOwnPropertyNames()`, `Object.getOwnPropertySymbols()`, `Object.getPrototypeOf()` - Behave similarly to when called on objects
+
+- `Object.prototype.toString`, `Object.prototype.toLocaleString` - Return `[object Record]` and `[object Tuple]`
+- `Object.prototype.valueOf` - Boxes a primitive record or tuple into an object
+- `Object.prototype.propertyIsEnumerable` - Always returns `true` for existing properties
+- `Object.prototype.hasOwnProperty`
+- `Object.prototype.isPrototypeOf`
+
+## `Object` methods which don't work with `Record` and `Tuple`
+
+- `Object.create()` - Records and tuples cannot have a prototype
```

Comment:
they could *be* a [[Prototype]], though, so I’m not sure how this entry makes sense - you certainly can’t produce a non-Object from an Object method, but you can pass a Record or Tuple into Object.create anywhere?

Reply:
Well, the current spec text would throw a TypeError. You'd have to do `Object.create(Object(#{}))` explicitly. I could go either way on whether we relax this constraint. (I do agree that the explanation doesn't make sense.)